plugins/gradle/gradle-plugin/src/main/java/org/wildfly/swarm/plugin/gradle/ThorntailArquillianPlugin.java

```diff
@@ -44,11 +58,65 @@ public ThorntailArquillianPlugin(ToolingModelBuilderRegistry registry) {
     @Override
     public void apply(Project project) {
         super.apply(project);
-        // Add the Gradle tooling dependency if it is missing.
-        Configuration testRuntimeConfig = project.getConfigurations().findByName("testRuntimeClasspath");
-        if (testRuntimeConfig != null) {
-            project.getDependencies().add("testRuntimeClasspath",
-                                          "org.gradle:gradle-tooling-api:" + project.getGradle().getGradleVersion());
+
+        //noinspection Convert2Lambda
+        project.afterEvaluate(new Action<Project>() {
+            @Override
+            public void execute(Project __) {
+                final String CONFIGURATION = "testRuntimeClasspath";
+                // Add the Gradle tooling dependency if it is missing.
+                String gav = "org.gradle:gradle-tooling-api:" + project.getGradle().getGradleVersion();
+                Configuration config = project.getConfigurations().findByName("testRuntimeClasspath");
+                if (config != null) {
+                    DependencyHandler handler = project.getDependencies();
+                    handler.add(config.getName(), handler.create(gav));
+                } else {
+                    System.err.println("Unable to add Gradle Tooling APIs to the " + CONFIGURATION +
+                            ". Thorntail Arquillian integration might not work.");
+                }
+
+                URL resource = ThorntailArquillianPlugin.class.getClassLoader().getResource(BUILD_SCRIPT_FRAGMENT);
+                if (resource != null) {
+                    try {
+                        File scriptFile = new File(project.getBuildDir(), "thorntail-arquillian-script.gradle");
+                        if (!scriptFile.exists()) {
+                            scriptFile.createNewFile();
+                            String[] lines = readString(resource.openStream()).split("\n");
+                            Files.write(scriptFile.toPath(), Arrays.asList(lines));
+                        }
+                        Map<String, Object> map = Collections.singletonMap("from", scriptFile.getAbsolutePath());
+                        project.apply(map);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                } else {
+                    System.err.println("Thorntail Arquillian:: Unable to locate resource: " + BUILD_SCRIPT_FRAGMENT);
+                }
+            }
+        });
+    }
+
+    /**
+     * Read the given stream in to a String. This method will close the stream as well.
+     *
+     * @param stream the input stream.
+     * @return the string built out of the data available in the given stream.
+     */
+    private static String readString(InputStream stream) throws IOException {
```

Comment:
Isn't it sad that there's no equivalent of `Files.readAllBytes` for classpath resources? :-/